src/attributes/limits.md

```diff
@@ -7,8 +7,42 @@ The following [attributes] affect compile-time limits.
 The *`recursion_limit` attribute* may be applied at the crate level to set the
 maximum depth for potentially infinitely-recursive compile-time operations
 like auto-dereference or macro expansion. It uses the [_MetaNameValueStr_]
-syntax to specify the recursion depth. The default is
-`#![recursion_limit="64"]`.
+syntax to specify the recursion depth. The default is 64.
+
+```rust,ignore
+#![recursion_limit = "4"]
+
+macro_rules! a {
+    () => { a!(1) };
+    (1) => { a!(2) };
+    (2) => { a!(3) };
+    (3) => { a!(4) };
+    (4) => { };
+}
+
+// This fails to expand because it requires a recursion depth greater than 4.
+a!{}
+```
+
+## The `type_length_limit` attribute
+
+The *`type_length_limit` attribute* limits the maximum size of a type
+constructed during monomorphization. It is applied at the crate level, and
+uses the [_MetaNameValueStr_] syntax to set the limit based on the number of
+type substitutions within the type. The default value is 1048576.
+
+```rust,ignore
+#![type_length_limit = "8"]
+
+type A = (B, B, B);
+type B = (C, C, C);
+struct C;
+
+// This fails to compile because monomorphizing to
+// `drop::<Option<((C, C, C), (C, C, C), (C, C, C))>>` requires more than 8
+// type elements.
+drop::<Option<A>>(None);
```

Comment:
Needing Turbofish syntax and `drop` here makes this example a bit more confusing. I'd show an identity function or something equivalent.

Reply:
I'm not sure I fully understand. Do you mean something like this?

```rust
#![type_length_limit = "8"]

fn f<T>(x: T) {}

// This fails to compile because monomorphizing to
// `f::<((i32, i32, i32), (i32, i32, i32), (i32, i32, i32))>>` requires more
// than 8 type elements.
f(((1, 2, 3), (4, 5, 6), (7, 8, 9)));
```

As an aside, I was uneasy about adding any examples at all, since they don't work with rustdoc, and it's relatively obscure, and could change in the future.

Reply:
That would work as an example.